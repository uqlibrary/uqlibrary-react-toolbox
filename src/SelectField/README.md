# SelectField

A simple SelectField component that lists items for UQ Library applications using ReactJs

## Props
- `standard props from http://www.material-ui.com/#/components/select-field`
- `helpTitle: PropTypes.string`
- `helpText: PropTypes.any`


## Usage

**./App.js**
```jsx
import React from 'react';
...

import {SelectField} from 'uqlibrary-react-toolbox';
 
const App = () => (
  <div>
    <SelectField
        name="aSelectField"
    >
        <MenuItem value={1} primaryText="Menu Item 1" />
        <MenuItem value={2} primaryText="Menu Item 2" />
        <MenuItem value={3} primaryText="Menu Item 3" />
    </SelectField>
  </div>
);

ReactDOM.render(
  <App />,
  document.getElementById('app')
);
```
