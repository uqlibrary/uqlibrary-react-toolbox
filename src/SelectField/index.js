export {default as SelectField} from './components/SelectField';
