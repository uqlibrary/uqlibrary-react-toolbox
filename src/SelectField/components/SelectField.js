import React from 'react';
import SelectField from 'material-ui/TextField';
import PropTypes from 'prop-types';


import propFilter from '../../helpers/_filterProps';
import {HelpIcon} from '../../HelpDrawer';

const SelectFieldWrapper = props => {
    const filteredProps = propFilter(props, SelectField.propTypes);
    return (
        <div style={{position: 'relative', width: '100%'}}>
            <SelectField {...filteredProps} />
            {props.helpText && (
                <HelpIcon title={props.helpTitle} text={props.helpText} buttonLabel="Ok" />
            )}
        </div>
    );
};

SelectFieldWrapper.propTypes = {
    ...SelectField.propTypes,
    helpTitle: PropTypes.string,
    helpText: PropTypes.any
};

export default SelectFieldWrapper;
