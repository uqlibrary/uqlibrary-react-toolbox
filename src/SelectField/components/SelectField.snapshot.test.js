jest.dontMock('./SelectField');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import SelectField from './SelectField';

import filterProps from '../../helpers/_filterProps';

function setup(props) {
    const consolidatedProps = filterProps(props, SelectField.propTypes);
    return shallow(<SelectField {...consolidatedProps} />);
}

describe('SelectfieldWrapper snapshots tests', () => {
    it('renders SelectField component', () => {
        const props =
            {
                name: 'selectfield',
                type: 'text',
                fullWidth: true,
                floatingLabelText: 'This is a test selectfield component'
            };

        const wrapper = setup(props);
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
