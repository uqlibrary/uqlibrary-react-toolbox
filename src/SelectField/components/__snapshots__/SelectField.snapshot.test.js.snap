// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectfieldWrapper snapshots tests renders SelectField component 1`] = `
<div
  style={
    Object {
      "position": "relative",
      "width": "100%",
    }
  }
>
  <SelectField
    autoWidth={false}
    disabled={false}
    floatingLabelText="This is a test selectfield component"
    fullWidth={true}
    multiple={false}
    name="selectfield"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
</div>
`;
